INSTALL-ADVANCED.md

```diff
@@ -0,0 +1,137 @@
+# balenaCLI Advanced Installation Options
```

Comment:
Before this PR, there was only `INSTALL.md`, which covered all installation options (GUI installer, standalone zip, npm) for all supported operating systems (Mac, Windows, Linux). This was considered confusing to many users. To simplify the instructions for the most common installation options, this PR proposes the following restructuring:

* The instructions are separated by OS:
  * INSTALL-WINDOWS.md for basic/recommended Windows instructions.
  * INSTALL-MAC.md for basic/recommended macOS instructions.
  * INSTALL-LINUX.md for basic/recommended Linux instructions.
  * INSTALL-ADVANCED.md for advanced options such as the npm installation, and "extra details" for advanced users.

* INSTALL.md, the original file, to be very short and mainly contain links to the files above.

Note this is still way short of the ultimate goal of have a direct link on the Getting Started guide, and reducing the instructions overall to "download, install, run" without needing to document additional dependencies and the like. But hopefully it is an improvement compared to what we have today.

Reply:
Awesome work @pdcastro maybe the new folks can do some [friction logging ](https://devrel.net/developer-experience/an-introduction-to-friction-logging) for us on this and see if they feel an improvement.